File: app/src/main/java/com/billpayments/core/event/UiEvent.kt
package com.billpayments.core.event

sealed interface UiEvent {
    data class PaymentStarted(val paymentId: String) : UiEvent
    data class PaymentFailed(val paymentId: String) : UiEvent
    data object OtpRejected : UiEvent
    data object AuthFailed : UiEvent
    data object StorageFailed : UiEvent
}

File: src/core/event/uiEvent.ts
export type UiEvent =
  | { readonly kind: 'paymentStarted'; readonly paymentId: string }
  | { readonly kind: 'paymentFailed'; readonly paymentId: string }
  | { readonly kind: 'otpRejected' }
  | { readonly kind: 'authFailed' }
  | { readonly kind: 'storageFailed' };
